libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -210,7 +217,7 @@ void AP_ICEngine::update(void)
                 // reset initial height while disarmed
                 initial_height = -pos.z;
             }
-        } else {
+        } else if (idle_percent <= 0) { // check if we should idle
```

Comment:
I'm having second thoughts about this one. This says if you have an idle pct, then you'll NEVER be able to turn the motor off, even if you disarm. That can't be right. How would you turn the motor off?